### DISCUSSION TITLE
How to close current tab (not window)?

### DISCUSSION THREAD

**Discussion by BartWeb3dev** [kaliiiiiiiiii/Selenium-Driverless#296] [Q&A] [answered]
Hi, how to close current tab? i wanna build small scarper on raspbbery pi andit has limit performance so i think it will be better way to set everything in 1 window than many windows. In docs i just found how to close current window (await driver.close())

**Comment by ttlns** [comment]
`driver.current_target.close`
Don't forget that there has to be at least one tab at a time (afaik)

**Reply by BartWeb3dev** [reply]
Thx for fast answer! It still close window  :/  What should i change?
```
        try:
            container = await driver.find_element(By.XPATH, '/html/body/div[1]/div/main/div/div[2]/div[4]', timeout=10)
            links = await container.find_elements(By.TAG_NAME, 'a')

            for link in links:
                href = await link.get_attribute('href')
                if href:
                    print("Opening links in new tab:", href)
                    await driver.new_window(type_hint='tab', url=href, activate=True)

                    # FUTURE CODE #
                    await driver.sleep(5)
                    await driver.current_target.close()

```

**Reply by ttlns** [reply, marked as answer]
```python
target = await driver.new_window(type_hint='tab', url=href, activate=True)
await target.close()
```
It's literally in the documentation haha

**Reply by BartWeb3dev** [reply]
> ```python
> target = await driver.new_window(type_hint='tab', url=href, activate=True)
> await target.close()
> ```
> 
> It's literally in the documentation haha

Yeah i read your docs but it's still close browser :/ idk how to resolve that Have you piece of code where you open a few tab and close some of them?

**Reply by BartWeb3dev** [reply]
It started working today idk why i didnt change the code. So yeah its should work as it described
